packages/client-sdk-nodejs/src/internal/control-client.ts

```diff
@@ -168,9 +172,23 @@ export class ControlClient {
           if (err || !resp) {
             resolve(new ListCaches.Error(cacheServiceErrorMapper(err)));
           } else {
-            const caches = resp.cache.map(
-              cache => new CacheInfo(cache.cache_name)
-            );
+            const caches = resp.cache.map(cache => {
+              const cacheName = cache.cache_name;
+              const topicLimits = new TopicLimits({
+                maxPublishMessageSizeKb:
+                  cache.topic_limits?.max_publish_message_size_kb || 0,
+                maxSubscriptionCount:
+                  cache.topic_limits?.max_subscription_count || 0,
+                maxPublishRate: cache.topic_limits?.max_publish_rate || 0,
+              });
+              const cacheLimits = new CacheLimits({
+                maxTtlSeconds: cache.cache_limits?.max_ttl_seconds || 0,
+                maxItemSizeKb: cache.cache_limits?.max_item_size_kb || 0,
+                maxThroughputKbps: cache.cache_limits?.max_throughput_kbps || 0,
+                maxTrafficRate: cache.cache_limits?.max_traffic_rate || 0,
+              });
```

Comment:
just checking my understanding of the proto changes - `cache.cache_name` is always present but the `cache_limits` and `topics_limits` were modeled as optional?

Reply:
they are modeled as always present, but until the change gets all the way through the backend, they will be returned as null

Reply:
but I'm surprised the TS compiler will accept this? If the type is defined as non-null?

Reply:
okay. interesting.

worth capturing a ticket to come back and remove the `?` later?

Reply:
yea, the other weird thing about this is that the grpc web generated code compiles in ts to show that its optional. The js generated code says its required

Reply:
so the grpc web code will always need a `?`

Reply:
why is it different for `cache_name` though?

Reply:
I believe because it is a grpc object vs a string, but I am not entirely sure

Reply:
okay. it's probably fine like this. thanks.

Reply:
ill make a ticket to remove the `?` for the nodejs sdk tho

Reply:
https://github.com/momentohq/client-sdk-javascript/issues/755